src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -128,7 +132,17 @@ export class TechRecordSummaryComponent implements OnInit {
   }
 
   calculateVehicleModel(): void {
-    this.vehicleTechRecordCalculated = this.isEditing ? { ...cloneDeep(this.vehicleTechRecord), reasonForCreation: '' } : this.vehicleTechRecord;
+    this.isEditing
+      ? this.technicalRecordService.editableTechRecord$
+          .pipe(
+            //Need to check that the editing tech record has more than just reason for creation on and is the full object.
+            map(data => (data && Object.keys(data).length > 1 ? cloneDeep(data) : { ...cloneDeep(this.vehicleTechRecord), reasonForCreation: '' })),
```

Comment:
```suggestion
            map(data => (data && Object.keys(data).length ? {...cloneDeep(data), reasonForCreation: data.reasonForCreation ?? ''} : { ...cloneDeep(this.vehicleTechRecord), reasonForCreation: '' })),
```

So if the comment from @naathanbrown on this same area from before about having to initialise the reasonForCreation is still valid, I believe something like this might need to be done.

Also presumably checking it has more than 0 keys (so a truthy value would be enough) unless saying there is more than one key is the desired behaviour

Reply:
No greater than 1 is what I want. If you refresh, you lose the tech record part of editing tech record, BUT you keep the reason for creation because that is added in the method above. So that observable ends up returning true and breaking the page because it cannot load in the other data. by adding this check, you know the actual tech record is present and not just the reason for creation. I appreciate this is silly but I'm pretty sure we've scoped time to refactor all of this at some point so ideally I'd like to leave it as is now.

Reply:
Really what this is checking is that data > data.reasonForCreation, and the easiest way to do that is say do I have more than 1 property